src/box/relay.cc

```diff
@@ -201,13 +204,20 @@ relay_new(struct replica *replica)
 	 * (Use clang UB Sanitizer, to make sure of this)
 	 */
 	assert((sizeof(struct relay) % alignof(struct relay)) == 0);
-	struct relay *relay = (struct relay *)
-		aligned_alloc(alignof(struct relay), sizeof(struct relay));
-	if (relay == NULL) {
+	/*
+	 * According to posix_memalign requirements, align must be
+	 * exponent of two and a multiple of sizeof(void *).
+	 */
+	assert(is_exp_of_two(alignof(struct relay)) &&
```

Comment:
Isn't this check inlined  into posix_memalign? What happens if align passed to posix_memalign is not power of two?

Reply:
In this case posix_memaligned return EINVAL and didn't allocate memory

Reply:
Anyway, I'd remove this assertion. Let's see what other reviewers will say.

Reply:
`alignof` is always power of two. should not be checked.
`mutliple of sizeof(void *)` should be checked in compile-time, not in assert.